File: src/Java/BehavioralPatterns/Command/FastFoodRestaurant/src/Patron.java
import Commands.FastFoodOrder;
import Commands.OrderCommand;
import Models.MenuItem;

public class Patron {
    private OrderCommand orderCommand;
    private MenuItem menuItem;
    private final FastFoodOrder fastFoodOrder;

    public Patron() {
        fastFoodOrder = new FastFoodOrder();
    }

    public void setCommand(int commandOption) {
        orderCommand = new CommandFactory().getCommand(commandOption);
    }

    public void setMenuItem(MenuItem menuItem) {
        this.menuItem = menuItem;
    }

    public void executeFastFoodOrderCommand() {
        fastFoodOrder.executeCommand(orderCommand, menuItem);
    }

    public void showCurrentOrder() {
        fastFoodOrder.showCurrentItems();
    }
}

File: src/C#/BehavioralPatterns/Command/FastFoodRestaurant/Patron.cs
using FastFoodRestaurant.Commands;
using FastFoodRestaurant.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace FastFoodRestaurant
{
    public class Patron
    {
        private OrderCommand orderCommand;

        private MenuItem menuItem;

        private readonly FastFoodOrder fastFoodOrder;

        public Patron()
        {
            fastFoodOrder = new FastFoodOrder();
        }

        public void SetCommand(int commandOption)
        {
            orderCommand = new CommandFactory().GetCommand(commandOption);
        }

        public void SetMenuItem(MenuItem menuItem)
        {
            this.menuItem = menuItem;
        }

        public void ExecuteFastFoodOrderCommand()
        {
            fastFoodOrder.ExecuteCommand(orderCommand, menuItem);
        }

        public void ShowCurrentOrder()
        {
            fastFoodOrder.ShowCurrentItems();
        }
    }
}
